gp-includes/routes/profile.php

```diff
@@ -16,34 +16,13 @@ public function profile_get() {
 		$this->tmpl( 'profile' );
 	}
 
-	public function profile_post( $user_id = null ) {
-		if ( isset( $_POST['submit'] ) ) {
-			// Sometimes we get null, sometimes we get 0, depending on where it comes from.
-			// Let's make sure we have a consistent value to test against and that it's an integer.
-			$user_id = (int) $user_id;
-
-			if ( 0 === $user_id ) { 
-				$user_id = get_current_user_id(); 
-			}
-			
-			$per_page = (int) $_POST['per_page'];
-			update_user_option( $user_id, 'gp_per_page', $per_page );
-
-			$default_sort = array(
-				'by'  => 'priority',
-				'how' => 'desc'
-			);
-			
-			$user_sort = wp_parse_args( $_POST['default_sort'], $default_sort );
-			update_user_option( $user_id, 'gp_default_sort', $user_sort );
+	public function profile_view( $user = '' ) {
+		if ( '' == $user ) { 
+			$user = wp_get_current_user(); 
+		} else {		
+			$user = get_user_by( 'slug', $user );
 		}
 
-		$this->redirect( gp_url( '/profile' ) );
```

Comment:
IMO `/profile` should redirect to `/profile/$current_user`. In case of a non-logged in access return a 404.